go/vt/vtctld/vttablet_proxy.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2020 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vtctld
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+	"sync"
+
+	"vitess.io/vitess/go/vt/log"
+)
+
+var (
+	proxyTablets = flag.Bool("proxy_tablets", false, "Setting this true will make vtctld proxy the tablet status instead of redirecting to them")
+
+	proxyMu sync.Mutex
+	remotes = make(map[string]*httputil.ReverseProxy)
+)
+
+func addRemote(tabletID, path string) {
+	u, err := url.Parse(path)
+	if err != nil {
+		log.Errorf("Error parsing URL %v: %v", path, err)
+		return
+	}
+
+	proxyMu.Lock()
+	defer proxyMu.Unlock()
+	if _, ok := remotes[tabletID]; ok {
+		return
```

Comment:
Does this mean the proxy will stop working for a given tablet after its address changes?